### PR TITLE
Finalize segment in LL mode on end_of_stream

### PR DESCRIPTION
**Pull request by Qizot**
The implementation of low-latency mode didn't finalize the last last segment once the `end_of_stream` has been received, resulting in the last regular segment getting lost.

### CODE REVIEW

File: lib/membrane_http_adaptive_stream/sink.ex [hunk 1]
```diff
@@ -450,6 +435,61 @@ defmodule Membrane.HTTPAdaptiveStream.Sink do
...
+  end
+
+  defp maybe_finalize_and_store_segment(track_id, state) do
+    %{manifest: manifest, storage: storage} = state
+
+    {partials, partials_duration} = partial_segments_for_track(track_id, state)
+
+    if partials != [] do
+      with {segment, changeset, manifest} <-
+             finalize_segment_for_track(track_id, partials, partials_duration, manifest),
```
Comment (mat-hek) [inline lib/membrane_http_adaptive_stream/sink.ex:477]: this will always match

File: lib/membrane_http_adaptive_stream/sink.ex [hunk 2]
```diff
@@ -271,8 +271,10 @@ defmodule Membrane.HTTPAdaptiveStream.Sink do
 
   @impl true
   def handle_end_of_stream(Pad.ref(:input, id), _ctx, state) do
-    %{manifest: manifest, storage: storage} = state
+    {storage, manifest} = maybe_finalize_and_store_segment(id, state)
```
Comment (mat-hek) [inline lib/membrane_http_adaptive_stream/sink.ex:274]: I'd put the contents of this function here

**Review by mat-hek — CHANGES_REQUESTED** [review]

File: lib/membrane_http_adaptive_stream/sink.ex (hunk 2 above)
Comment (mat-hek) [reply lib/membrane_http_adaptive_stream/sink.ex:274]: Ok, the main problem is that the name `maybe_finalize_and_store_segment` doesn't suggest it's only valid for end_of_stream. It took me a while to realize why `handle_partial_segment` doesn't use it.

File: lib/membrane_http_adaptive_stream/sink.ex [hunk 3]
```diff
@@ -450,6 +435,59 @@ defmodule Membrane.HTTPAdaptiveStream.Sink do
...
+      case Storage.apply_segment_changeset(storage, track_id, changeset, segment) do
+        {:ok, storage} ->
+          {storage, manifest}
+
+        {{:error, reason}, _storage} ->
+          raise reason
+      end
+    else
+      {storage, manifest}
+    end
```
Comment (mat-hek) [inline lib/membrane_http_adaptive_stream/sink.ex:488]:
```suggestion
    if partials == [] do
      {storage, manifest}
    else
      {segment, changeset, manifest} =
        finalize_segment_for_track(track_id, partials, partials_duration, manifest)

      case Storage.apply_segment_changeset(storage, track_id, changeset, segment) do
        {:ok, storage} -> {storage, manifest}
        {{:error, reason}, _storage} -> raise reason
      end
    end
```

Maybe it'll be a bit more readable this way?

File: lib/membrane_http_adaptive_stream/sink.ex (hunk 2 above)
Comment (Qizot) [reply lib/membrane_http_adaptive_stream/sink.ex:274]: I don't want to do that java styles but wdyt about `maybe_finalize_and_store_segment_on_eos(...)`? (`eos` so the name doesn't become a poem).

**Review by mat-hek — APPROVED** [review]

**Pull request merged**